### PR TITLE
Initial work on setup.cfg and tidy tox

### PR DESCRIPTION
**Pull request by stufisher**
Move metadata out of `setup.py` into `setup.cfg`. Remove flake8 configuration from `tox.ini`. Add flake8 rules to allow using database column names as functions and arguments (https://github.com/ispyb/py-ispyb/pull/135). Disable flake8 rules to make it play nice with black (https://github.com/ispyb/py-ispyb/pull/136)

Closes https://github.com/ispyb/py-ispyb/issues/132, https://github.com/ispyb/py-ispyb/issues/130

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    Programming Language :: Python :: 3.10
+    Operating System :: OS Independent
+    Topic :: Software Development :: Libraries :: Python Modules
+project-urls =
+    download = https://github.com/ispyb/py-ispyb
+    documentation = https://ispyb.gitlab-pages.esrf.fr/py-ispyb/
+
+
+[options]
+python_requires = >=3.9
```
Comment (mgaonach) [inline setup.cfg:25]: I think we need Python >= 3.10 as we use some of the Python 3.10 notation for typing in the code

**Comment by stufisher** [comment]
Mostly supersedes https://github.com/ispyb/py-ispyb/pull/135 (for now). I didn't yet apply any additional flake8 plugins so it is not enforcing function and parameter naming. We can do that in another mr. Sound ok @mgaonach ?

File: setup.cfg (hunk 1 above)
Comment (stufisher) [reply setup.cfg:25]: yep good spot

**Comment by mgaonach** [comment]
Looks good!

**Review by mgaonach — APPROVED** [review]

**Pull request merged**